src/gen_rules.ml

```diff
@@ -164,51 +164,44 @@ module Gen(P : Install_rules.Params) = struct
       else
         `Regular
     in
-    let regular_rule run_action alias loc =
-      { alias with Alias_conf.action = Some (loc, run_action) }
-    in
-    let expect_rule run_action (diff : Action.Unexpanded.Diff.t) alias loc =
-      let rule =
-        { Rule.
-          targets = Infer
-        ; deps = Bindings.empty
-        ; action =
-            (loc, Action.Unexpanded.Redirect (Stdout, diff.file2, run_action))
-        ; mode = Standard
-        ; locks = t.locks
-        ; loc
-        } in
-      let alias =
-        { alias with
-          Alias_conf.
-          action = Some (loc, Diff diff)
-        ; locks = t.locks
-        } in
-      (alias, rule)
-    in
     List.iter t.exes.names ~f:(fun (loc, s) ->
+      let test_exe = "./" ^ s ^ ".exe" in
       let run_action =
-        Action.Unexpanded.Run
-          (String_with_vars.make_text loc ("./" ^ s ^ ".exe"), []) in
-      let base_alias =
-        { Alias_conf.
-          name = "runtest"
-        ; locks = []
-        ; package = t.package
-        ; deps = t.deps
-        ; action = None
-        ; enabled_if = t.enabled_if
-        ; loc
-        } in
+        match t.action with
+        | Some a -> a
+        | None -> Action.Unexpanded.Run (String_with_vars.make_text loc test_exe, [])
+      in
+      let extra_bindings = Pform.Map.singleton "test" (Pform.Var.Values [Value.String test_exe]) in
```

Comment:
The value should be a `Path` so that it expands to the right path if the user does: `(chdir %{project_root} (run %{test}))`. Additionally, it should resolve to the executable in the host build context in case of cross-compilation (the function to do that is `Super_context.Action.map_exe`).